spring-pulsar/src/main/java/org/springframework/pulsar/annotation/ReactivePulsarListener.java

```diff
@@ -0,0 +1,156 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.apache.pulsar.client.api.SubscriptionType;
+import org.apache.pulsar.common.schema.SchemaType;
+
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.pulsar.config.reactive.ReactivePulsarListenerContainerFactory;
+import org.springframework.pulsar.config.reactive.ReactivePulsarListenerEndpointRegistry;
+
+/**
+ * Annotation that marks a method to be the target of a Pulsar message listener on the
+ * specified topics.
+ *
+ * The {@link #containerFactory()} identifies the
+ * {@link ReactivePulsarListenerContainerFactory} to use to build the Pulsar listener
+ * container. If not set, a <em>default</em> container factory is assumed to be available
+ * with a bean name of {@code pulsarListenerContainerFactory} unless an explicit default
+ * has been provided through configuration.
+ *
+ * <p>
+ * Processing of {@code @ReactivePulsarListener} annotations is performed by registering a
+ * {@link ReactivePulsarListenerAnnotationBeanPostProcessor}. This can be done manually
+ * or, more conveniently, through {@link EnablePulsar} annotation.
+ * </p>
+ *
+ * @author Christophe Bornet
+ */
+@Target({ ElementType.TYPE, ElementType.METHOD, ElementType.ANNOTATION_TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+@MessageMapping
+@Documented
+public @interface ReactivePulsarListener {
+
+	/**
+	 * The unique identifier of the container for this listener.
+	 * <p>
+	 * If none is specified an auto-generated id is used.
+	 * <p>
+	 * SpEL {@code #{...}} and property place holders {@code ${...}} are supported.
+	 * @return the {@code id} for the container managing for this endpoint.
+	 * @see ReactivePulsarListenerEndpointRegistry#getListenerContainer(String)
+	 */
+	String id() default "";
+
+	/**
+	 * Pulsar subscription name associated with this listener.
+	 * @return the {@code subscriptionName} for this Pulsar listener endpoint.
+	 */
+	String subscriptionName() default "";
+
+	/**
+	 * Pulsar subscription type for this listener.
+	 * @return the {@code subscriptionType} for this listener
+	 */
+	SubscriptionType subscriptionType() default SubscriptionType.Exclusive;
+
+	SchemaType schemaType() default SchemaType.NONE;
```

Comment:
**[NIT]** Do you mind adding javadoc here (as well as the imperative one that was already missing it)?

Reply:
done